xarray/core/variable.py

```diff
@@ -761,7 +757,14 @@ def __getitem__(self, key) -> Self:
         array `x.values` directly.
         """
         dims, indexer, new_order = self._broadcast_indexes(key)
-        data = as_indexable(self._data)[indexer]
+        indexable = as_indexable(self._data)
+
+        if isinstance(indexer, BasicIndexer):
+            data = indexable[indexer]
+        elif isinstance(indexer, OuterIndexer):
+            data = indexable.oindex(indexer)
+        else:
+            data = indexable.vindex(indexer)
```

Comment:
We should be passing the underlying tuple down to the indexing wrappers
```suggestion
        if isinstance(indexer, BasicIndexer):
            data = indexable[indexer.key]
        elif isinstance(indexer, OuterIndexer):
            data = indexable.oindex(indexer.key)
        else:
            data = indexable.vindex(indexer.key)
```

Reply:
@dcherian, i looked into this, and it turned out that passing the underlying tuple at this stage (without support for `.vindex`) requires a series of special cases for orthogonal indexing. this results in extensive code duplication between the basic indexer and the vectorized indexer. it might be easier to address this in a separate PR once `vindex` is in place